crates/server/src/store/in_memory/authority.rs

```diff
@@ -1201,22 +1200,31 @@ impl Authority for InMemoryAuthority {
             }
             RecordType::AXFR => {
                 // TODO: shouldn't these SOA's be secure? at least the first, perhaps not the last?
-                let lookup = future::try_join3(
-                    // TODO: maybe switch this to be an soa_inner type call?
-                    self.soa_secure(lookup_options),
-                    self.soa(),
-                    self.lookup(lookup_name, record_type, lookup_options),
-                )
-                .map_ok(|(start_soa, end_soa, records)| match start_soa {
-                    l @ AuthLookup::Empty => l,
-                    start_soa => AuthLookup::AXFR {
-                        start_soa: start_soa.unwrap_records(),
-                        records: records.unwrap_records(),
-                        end_soa: end_soa.unwrap_records(),
-                    },
-                });
-
-                lookup.await
+                let start_soa = if let LookupResult::Ok(res) = self.soa_secure(lookup_options).await
```

Comment:
This also looks like it has unrelated changes.

Reply:
This one is directly related to the type change.